diff --git a/libraries/classes/DatabaseInterface.php b/libraries/classes/DatabaseInterface.php
index 179285852d..2196b18d03 100644
--- a/libraries/classes/DatabaseInterface.php
+++ b/libraries/classes/DatabaseInterface.php
@@ -1548,7 +1548,7 @@ class DatabaseInterface
         /* Locale for messages */
         $locale = LanguageManager::getInstance()->getCurrentLanguage()->getMySQLLocale();
         if (! empty($locale)) {
-            $this->query(
+            $this->tryQuery(
                 "SET lc_messages = '" . $locale . "';",
                 DatabaseInterface::CONNECT_USER,
                 self::QUERY_STORE
