diff --git a/libraries/vendor_config.php b/libraries/vendor_config.php
index 669821025b..a2c1673a93 100644
--- a/libraries/vendor_config.php
+++ b/libraries/vendor_config.php
@@ -24,7 +24,7 @@ define('AUTOLOAD_FILE', ROOT_PATH . 'vendor/autoload.php');
 /**
  * Directory where cache files are stored.
  */
-define('TEMP_DIR', ROOT_PATH . 'tmp/');
+define('TEMP_DIR', getenv('PMA_TEMP_DIR') ?: ROOT_PATH . 'tmp/');
 
 /**
  * Path to changelog file, can be gzip compressed. Useful when you want to
@@ -48,7 +48,7 @@ define('SQL_DIR', ROOT_PATH . 'sql/');
  * It is not used directly in code, just a convenient
  * define used further in this file.
  */
-define('CONFIG_DIR', ROOT_PATH);
+define('CONFIG_DIR', getenv('PMA_CONFIG_DIR') ?: ROOT_PATH);
 
 /**
  * Filename of a configuration file.
