diff --git a/libraries/vendor_config.php b/libraries/vendor_config.php
index 62e1c1301e..dd1304a4d0 100644
--- a/libraries/vendor_config.php
+++ b/libraries/vendor_config.php
@@ -22,7 +22,7 @@
     /**
      * Directory where cache files are stored.
      */
-    'tempDir' => ROOT_PATH . 'tmp' . DIRECTORY_SEPARATOR,
+    'tempDir' => getenv('PMA_TEMP_DIR') ?: ROOT_PATH . 'tmp' . DIRECTORY_SEPARATOR,

     /**
      * Path to changelog file, can be gzip compressed.
@@ -43,7 +43,7 @@
     /**
      * Filename of a configuration file.
      */
-    'configFile' => ROOT_PATH . 'config.inc.php',
+    'configFile' => getenv('PMA_CONFIG_FILE') ?: ROOT_PATH . 'config.inc.php',

     /**
      * Filename of custom header file.
--
2.36.0
